File: DataStructures/LinkedList/DeleteLinkedListFunction.js
var head;

class Node {
    constructor(val)
    {
        this.data = val;
        this.next = null;
    }
}

function deleteList()
{
    document.write("List to be deleted: ");
    let tnode = head;
    while(tnode != null)
    {
        document.write(tnode.data + " ");
        tnode = tnode.next;
    }
    head = null;
    document.write("List deleted: ");
}

function push(new_data)
{
    let new_node = new Node(new_data);
    new_node.next = head;
    head = null;
}


push(1);
push(15);
push(13);
push(12);
push(10);
push(9);
push(11);
push(124);

deleteList();

File: DataStructures/LinkedList/DeleteLinkedListFunction.java
class LinkedList
{
    Node head;

    class Node 
    {
        int data;
        Node next;
        Node(int d)
        {
            data = d;
            next = null;
        }
    }

    void deleteList()
    {
        System.out.print("Deleting List : ");
        Node tnode = head;
        while(tnode != null)
        {
            System.out.print(tnode.data + " ");
            tnode = tnode.next;
        }
        head = null;
        System.out.print("List deleted");
    }
    // Function deletes the entire linked list

    public void push(int new_data)
    {
        Node new_node = new Node(new_data);

        new_node.next = head;
        head = new_node;
    }

    public static void main(String[] args)
    {
        LinkedList llist = new LinkdList();

        llist.push(1);
        llist.push(1);
        llist.push(1);
        llist.push(1);
        llist.deleteList();

    }
}